src/main/java/com/metamx/metrics/JvmThreadsMonitor.java

```diff
@@ -0,0 +1,56 @@
+package com.metamx.metrics;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.metamx.emitter.service.ServiceEmitter;
+import com.metamx.emitter.service.ServiceMetricEvent;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+import java.util.List;
+import java.util.Map;
+import org.gridkit.lab.jvm.perfdata.JStatData;
+import org.gridkit.lab.jvm.perfdata.JStatData.LongCounter;
+
+public class JvmThreadsMonitor extends AbstractMonitor
+{
+  private final Map<String, String[]> dimensions;
+
+  private int lastLiveThreads = 0;
+  private long lastStartedThreads = 0;
+
+  public JvmThreadsMonitor()
+  {
+    this(ImmutableMap.<String, String[]>of());
+  }
+
+  public JvmThreadsMonitor(Map<String, String[]> dimensions)
+  {
+    Preconditions.checkNotNull(dimensions);
+    this.dimensions = ImmutableMap.copyOf(dimensions);
+  }
+
+  @Override
+  public boolean doMonitor(ServiceEmitter emitter)
+  {
+    ThreadMXBean threadBean = ManagementFactory.getThreadMXBean();
+
+    final ServiceMetricEvent.Builder builder = new ServiceMetricEvent.Builder();
+    MonitorUtils.addDimensionsToBuilder(builder, dimensions);
+
+    int newLiveThreads = threadBean.getThreadCount();
+    long newStartedThreads = threadBean.getTotalStartedThreadCount();
+
+    long startedThreadsDiff = newStartedThreads - lastStartedThreads;
+
+    emitter.emit(builder.build("jvm/threads/daemon", threadBean.getDaemonThreadCount()));
+    emitter.emit(builder.build("jvm/threads/peak", threadBean.getPeakThreadCount()));
```

Comment:
Without `resetPeakThreadCount()` in the next line, semantic of this metric is different from the "live" metric semantic. Currently, "peak" is equivalent to having `private int maxLiveThreadObserved` field in this class, updating it on each `doMonitor()` call and emitting it instead of `newLiveThreads`.